packages/suite/src/middlewares/suite/suiteMiddleware.ts

```diff
@@ -114,6 +115,11 @@ const suite =
             api.dispatch(observeSelectedDevice());
         }
 
+        // Restart discovery to reconnect to backends when user goes offline -> online.
```

Comment:
You miss check going from offline to online 🧐

Reply:
This is how this action works based on this example & testing.
<img width="617" alt="image" src="https://github.com/user-attachments/assets/7a301b44-dd8d-46ed-af25-c312a0afcd20">

Reply:
Do I have to test it additionally?

Reply:
I think you have it correct. The online status is either true or false so it works both ways.

Reply:
Yes. It doesn't refresh on online -> offline though.

Reply:
Why would you restart it when going from online to offline? I don't get it 

Moreover, why it is not in the switch above and you introduce if statement

Reply:
It was not restarting due to how restartDiscoveryThunk worked, but I added the check anyway, so now it will restart only on going offline - online for sure